openedx/core/djangoapps/user_authn/views/tests/test_reset_password.py

```diff
@@ -123,21 +123,63 @@ def test_nonexist_email_password_reset(self):
         'openedx.core.djangoapps.user_authn.views.password_reset.render_to_string',
         Mock(side_effect=mock_render_to_string, autospec=True)
     )
-    def test_password_reset_ratelimited(self):
+    def test_password_reset_ratelimited_for_non_existing_user(self):
         """
-        Test that reset password endpoint only allow one request per minute.
+        Test that reset password endpoint only allow one request per minute
+        for non-existing user.
         """
-        cache.clear()
+        self.assert_password_reset_ratelimitted('thisdoesnotexist@foo.com', AnonymousUser())
+        self.assert_no_events_were_emitted()
+
+    @patch(
+        'openedx.core.djangoapps.user_authn.views.password_reset.render_to_string',
+        Mock(side_effect=mock_render_to_string, autospec=True)
+    )
+    def test_password_reset_ratelimited_for_existing_user(self):
+        """
+        Test that reset password endpoint only allow one request per minute
+        for existing user.
+        """
+        self.assert_password_reset_ratelimitted(self.user.email, self.user)
+        self.assert_event_emission_count(SETTING_CHANGE_INITIATED, 1)
 
-        password_reset_req = self.request_factory.post('/password_reset/', {'email': 'thisdoesnotexist@foo.com'})
-        password_reset_req.user = AnonymousUser()
+    def assert_password_reset_ratelimitted(self, email, user):
+        """
+        Assert that password reset endpoint allow one request per minute per email.
+        """
+        cache.clear()
+        password_reset_req = self.request_factory.post('/password_reset/', {'email': email})
+        password_reset_req.user = user
+        password_reset_req.site = Mock(domain='example.com')
         good_resp = password_reset(password_reset_req)
         self.assertEqual(good_resp.status_code, 200)
 
         # then the rate limiter should kick in and give a HttpForbidden response
         bad_resp = password_reset(password_reset_req)
         self.assertEqual(bad_resp.status_code, 403)
-        self.assert_no_events_were_emitted()
+
+        cache.clear()
+
+    @patch(
+        'openedx.core.djangoapps.user_authn.views.password_reset.render_to_string',
+        Mock(side_effect=mock_render_to_string, autospec=True)
+    )
+    def test_ratelimitted_from_same_ip_with_different_email(self):
```

Comment:
Thank you for adding this test. I do think that we should also test the case where users on two different IPs attempt to password reset the same email (i.e. `test_ratelimited_from_different_ips_with_same_email`).